### ISSUE TITLE
D-Pad & Select Quest controller mapping

### ISSUE THREAD

**Issue by ElBatallitas** [amwatson/CitraVR#19]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

An option to shortcut that is click left Thumbstick on Quest controller, to switch between analog pad and d-pad.
Maybe "hold" Quest "menu button" to input 3ds' "select"

### Why would this feature be useful?

so we don't need a gamepad.

**Comment by cyperium** [comment]
Holding the Quest "menu button" would make it impossible to press select + start at the same time, which is needed for button mapping. Instead maybe it would be best if clicking the left Thumbstick would be "select" and clicking the right Thumbstick would toggle between analog pad and d-pad?

**Comment by ElBatallitas** [comment]
> Holding the Quest "menu button" would make it impossible to press select + start at the same time, which is needed for button mapping. Instead maybe it would be best if clicking the left Thumbstick would be "select" and clicking the right Thumbstick would toggle between analog pad and d-pad?

That suggested mapping would work for me too. 
I didn't get the first part tho. How are you able to achieve "select + start" in quest now anyway? 
Isn't that combo meant for a gamepad.

**Comment by cyperium** [comment]
Yes, but shouldn't it be able to be simulated if we bind the thumbstick click to the select button? I don't know really how it works under the hood of Citra but obviously we need to be able to set the buttons for the Quest controllers, the only difference from a Gamepad that keeps it from working would be that it doesn't have a select button so we can't use the button combo that Citra detects. Another option is just to make another menu option for the Quest specifically I guess where you don't need to press that combo.

**Comment by Rose22** [comment]
the start+select thing isn't a thing on the quest controllers, it's specific to external bluetooth gamepads and is to toggle the gamepad between pointer mode and gaming mode. in gaming mode an external gamepad works in flat android apps, but in pointer mode it lets you control the quest UI by turning your head and pressing the A button.

so you dont need special mapping for the select button on quest controllers, but a way to switch between thumbstick and d-pad on the left stick is a good idea

**Comment by cyperium** [comment]
Ok, I didn't know that, wouldn't it be good to have a select "binding" though as some games might require using select? But yeah, the main feature that I would want is to be able to map the buttons on the Quest controller.
